skills/declarative/calendarSkill/dialogs/SearchEventsDialog/SearchEventsDialog.dialog

```diff
@@ -0,0 +1,373 @@
+                "id": "YTg9fk"
+              },
+              "condition": "dialog.turnCount",
+              "cases": [
+                {
+                  "value": "0",
+                  "actions": [
+                    {
+                      "$kind": "Microsoft.RepeatDialog",
+                      "$designer": {
+                        "id": "mz6WpY"
+                      },
+                      "activityProcessed": true,
+                      "options": {
+                        "turnCount": 1,
+                        "query": "repeat",
+                        "intent": "=dialog.intent"
+                      }
+                    }
+                  ]
+                },
+                {
+                  "value": "1",
+                  "actions": [
+                    {
+                      "$kind": "Microsoft.SendActivity",
+                      "$designer": {
+                        "id": "wyiHTF"
+                      },
+                      "activity": "${SendActivity_wyiHTF()}"
+                    },
+                    {
+                      "$kind": "Microsoft.RepeatDialog",
+                      "$designer": {
+                        "id": "8AwLVe"
+                      },
+                      "activityProcessed": true,
+                      "options": {
+                        "turnCount": 2,
+                        "query": "today",
+                        "intent": "=dialog.intent"
+                      }
+                    }
+                  ]
+                }
+              ],
+              "default": [
+                {
+                  "$kind": "Microsoft.SendActivity",
+                  "$designer": {
+                    "id": "vOa1jo"
+                  },
+                  "activity": "${SendActivity_vOa1jo()}"
+                }
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "$kind": "Microsoft.TraceActivity",
+              "$designer": {
+                "id": "PzalaP"
+              },
+              "name": "AfterSort"
+            },
+            {
+              "$kind": "Microsoft.Foreach",
+              "$designer": {
+                "id": "JFoghi"
+              },
+              "index": "dialog.foreach.index",
+              "value": "dialog.foreach.value",
+              "itemsProperty": "dialog.events",
+              "actions": [
+                {
+                  "$kind": "Microsoft.SetProperties",
+                  "$designer": {
+                    "id": "M5nVxN"
+                  },
+                  "assignments": [
+                    {
+                      "property": "dialog.events[dialog.foreach.index].Location",
+                      "value": "= dialog.events[dialog.foreach.index].Location.DisplayName"
+                    }
+                  ]
+                }
+              ]
+            },
+            {
+              "$kind": "Microsoft.IfCondition",
```

Comment:
The dialog should be able to handle returning either a list of events or a single event depending on how its configured in dialog options. So if you have a flag called "doSelection" for example you could show the choice prompt if its true, or return the full results list if its false. That way we could reuse this dialog for the Day Overview as well

Reply:
Sure, modified the flow to match this pattern.